src/com/facebook/buck/intellij/ideabuck/src/com/facebook/buck/intellij/ideabuck/actions/BuckRunSelectedTestAction.java

```diff
@@ -0,0 +1,89 @@
+package com.facebook.buck.intellij.ideabuck.actions;
+
+import com.facebook.buck.intellij.ideabuck.configurations.TestConfiguration;
+import com.facebook.buck.intellij.ideabuck.configurations.TestConfigurationType;
+import com.facebook.buck.intellij.ideabuck.file.BuckFileUtil;
+import com.intellij.execution.Executor;
+import com.intellij.execution.ExecutorRegistry;
+import com.intellij.execution.ProgramRunnerUtil;
+import com.intellij.execution.RunManager;
+import com.intellij.execution.impl.RunManagerImpl;
+import com.intellij.execution.impl.RunnerAndConfigurationSettingsImpl;
+import com.intellij.openapi.actionSystem.*;
+import com.intellij.openapi.editor.Document;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.fileEditor.FileDocumentManager;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethod;
+
+import java.io.File;
+
+public class BuckRunSelectedTestAction extends AnAction {
+
+  @Override
+  public void actionPerformed(final AnActionEvent event) {
+    PsiElement element = event.getData(LangDataKeys.PSI_ELEMENT);
+    PsiMethod method = (PsiMethod) element;
+    final Project project = event.getProject();
+    if (project == null) {
+      return;
+    }
+    Editor editor = FileEditorManager.getInstance(project).getSelectedTextEditor();
+    if (editor == null) {
+      return;
+    }
+    final Document document = editor.getDocument();
+    VirtualFile virtualFile = FileDocumentManager.getInstance().getFile(document);
+
+    VirtualFile buckFile = BuckFileUtil.getBuckFile(virtualFile);
+    if (buckFile != null) {
+      String relative = new File(project.getBasePath()).toURI().relativize(new File(buckFile.getParent().getCanonicalPath()).toURI()).getPath();
+      String target = relative+":test_main";
```

Comment:
This is still hardcoded at this point. Once I figure out the stale configurations it should be easy to wire the proper test target.